### PR TITLE
Speed powerup

### PR DESCRIPTION
**Pull request by jimhommes**
Because there is only 1 powerup, the random always chooses this powerup. So, when you pick up a powerup, your speed is increased for a few seconds.

<feature

### CODE REVIEW

File: src/main/java/model/Powerup.java [hunk 1]
```diff
@@ -75,15 +83,32 @@ public void causesCollision(SpriteBase player) {
...
             levelController.getScreenController().removeSprite(this);
             pickedUp = true;
             Logger.log("Picked up Powerup.");
+
+            if (kind < 1) {
+                for (int i = 0; i < AMOUNT_OF_POWERUPS; i++) {
+                    if (kind >= i * (kind / AMOUNT_OF_POWERUPS)) {
+                        player.activateSpeedPowerup();
+                    }
+                }
```
Comment (toinehartman) [inline src/main/java/model/Powerup.java]: I suggest `Math.ceil(kind*AMOUNT_OF_POWERUPS)` to check which powerup it is. The for loop can be removed.

<logic

File: src/main/java/model/Powerup.java [hunk 2]
```diff
@@ -75,15 +83,32 @@ public void causesCollision(SpriteBase player) {
...
+            }
+
         }
     }
 
+    private void activateSpeedPowerup() {
+        //TODO: Activate Speed Powerup
+    }
+
```
Comment (toinehartman) [inline src/main/java/model/Powerup.java:108]: Remove pls.

<organisation

**Comment by toinehartman** [comment]
:shipit: 

<feature

File: src/main/java/model/Powerup.java [hunk 3]
```diff
@@ -75,12 +83,23 @@ public void causesCollision(SpriteBase player) {
...
             pickedUp = true;
             Logger.log("Picked up Powerup.");
+
+            int k = (int) kind;
+
+            if (kind < 1) {
+                k = (int) Math.ceil(kind * AMOUNT_OF_POWERUPS);
+            }
+
+            switch (k) {
```
Comment (LilideBree) [inline src/main/java/model/Powerup.java:98]: Checkstyle error, coding, FallThrough. Probably the missing default case.
This is also a FindBugs error.

**Pull request merged**